numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -3305,8 +3316,6 @@ PyArray_Where(PyObject *condition, PyObject *x, PyObject *y)
 
     npy_intp itemsize = common_dt->elsize;
 
-    int has_ref = PyDataType_REFCHK(common_dt);
-
     NPY_ARRAYMETHOD_FLAGS transfer_flags = 0;
 
     npy_intp transfer_strides[2] = {itemsize, itemsize};
```

Comment:
```suggestion
    /* Single element copy; `itemsize, itemsize` is just a guess at contiguous. */
    npy_intp transfer_strides[2] = {itemsize, itemsize};
```
Just stumbled a second here because the first `itemsize` is now a random guess.  Maybe a comment makes sense (not that it changed and I am not sure).  Or just change it to `0, 0` I doubt it matters in practice.